eclipse/che
# best effort to label namespace when starting new workspace

Signed-off-by: Michal Vala <mvala@redhat.com>



### What does this PR do?
When starting new workspace, try label the namespace, if che SA have permissions to do so.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/17504

### How to test this PR?

1. Ensure that che SA has ClusterRole with `get` and `update` `namespace` permission. You can use this:
```
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: che-extraperm
rules:
- apiGroups: [""]
  resources: ["namespaces"]
  verbs: ["update", "get"]
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: che-extraperm
subjects:
- kind: ServiceAccount
  name: che
  namespace: che
roleRef:
  kind: ClusterRole
  name: che-extraperm
  apiGroup: rbac.authorization.k8s.io
```
2. Create a new workspace
3. Check that namespace where workspace has started is labeled with labels from `che.infra.kubernetes.namespace.labels` property

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
